File: src/main/java/com/aiproxyoauth/util/ProxyVersion.java
package com.aiproxyoauth.util;

import java.io.IOException;
import java.io.InputStream;
import java.util.Properties;

/**
 * Resolves the proxy's version from the {@code aiproxyoauth-version.properties} resource
 * that the build generates from the Gradle {@code pluginVersion} property, so every
 * surface (health endpoint, CLI --version) reports the plugin version. Running without
 * the generated resource (e.g. straight from an IDE compile) yields "dev".
 */
public final class ProxyVersion {

    private static final String VERSION_RESOURCE = "/aiproxyoauth-version.properties";
    private static final String DEV_VERSION = "dev";
    private static final String VERSION = resolve();

    private ProxyVersion() {}

    public static String get() {
        return VERSION;
    }

    private static String resolve() {
        try (InputStream in = ProxyVersion.class.getResourceAsStream(VERSION_RESOURCE)) {
            if (in != null) {
                Properties properties = new Properties();
                properties.load(in);
                String version = properties.getProperty("version");
                if (version != null && !version.isBlank()) {
                    return version.strip();
                }
            }
        } catch (IOException ignored) {
        }
        return DEV_VERSION;
    }
}

File: src/main/kotlin/de/moritzf/proxy/util/ProxyVersion.kt
package de.moritzf.proxy.util

import java.io.IOException
import java.util.Properties

/**
 * Resolves the proxy's version from the `aiproxyoauth-version.properties` resource
 * that the build generates from the Gradle `pluginVersion` property, so every surface
 * (health endpoint, CLI --version) reports the plugin version. Running without the
 * generated resource, e.g. straight from an IDE compile, yields "dev".
 */
object ProxyVersion {
    private const val VERSION_RESOURCE = "/aiproxyoauth-version.properties"
    private const val DEV_VERSION = "dev"
    private val version = resolve()

    @JvmStatic
    fun get(): String = version

    private fun resolve(): String {
        return try {
            ProxyVersion::class.java.getResourceAsStream(VERSION_RESOURCE)?.use { input ->
                val properties = Properties()
                properties.load(input)
                properties.getProperty("version")?.takeIf { it.isNotBlank() }?.trim()
            } ?: DEV_VERSION
        } catch (_: IOException) {
            DEV_VERSION
        }
    }
}
